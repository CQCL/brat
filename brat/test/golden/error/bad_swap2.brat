swap(X :: $, Y :: $) -> { X, Y -o Y, X }
swap(_, _) = { x, y => y, x }

call_swap(X :: $) -> { X, Qubit -o X, Qubit }
call_swap(X) = { x, q => swap(X, Qubit)(x, q) }
