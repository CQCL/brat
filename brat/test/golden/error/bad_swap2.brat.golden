Error in test/golden/error/bad_swap2.brat on line 5:
call_swap(X) = { x, q => swap(X, Qubit)(x, q) }
                         ^^^^^^^^^^^^^^^^^^^^

  Type mismatch when checking swap()(「X」, Qubit(()))()(「x」, 「q」)
Expected: (a1 :: X), (b1 :: Qubit)
But got:  (a1 :: Qubit), (b1 :: X)


